### PR TITLE
Hive integration test infra changes to run with version 3.x

### PR DESCRIPTION
**Pull request by nmahadevuni**
Cherry pick of following commits
https://github.com/trinodb/trino/commit/f90708111f50fc48a661e30bda49cca7f3d1abe8
https://github.com/trinodb/trino/commit/de0f2df5d2c592200477c3749ffcd4267d6dfe53
https://github.com/trinodb/trino/commit/802e2732b10b182f33d9a68250bc31d748c83aa9
https://github.com/trinodb/trino/commit/b8729fe4a117a65a273ca259844f9289e8d1700f
https://github.com/trinodb/trino/commit/5bfde980654e51bd946b1f820cbe5ad5ba803e28

Co-authored-by: Łukasz Osipiuk <lukasz@osipiuk.net>
Co-authored-by: Piotr Findeisen <piotr.findeisen@gmail.com>

Test plan - Ran the TestHiveClient.java tests using both prestodb/hdp2.6-hive and prestodb/hive3.1-hive images.

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Comment by nmahadevuni** [comment]
This PR contains changes to run the hive integration tests(TestHiveClient.java) on Hive 3.x. The docker image required for this has been created and pushed to docker hub at https://hub.docker.com/r/prestodb/hive3.1-hive/ for which changes were approved in PR https://github.com/prestodb/docker-images/pull/53.

**Comment by nmahadevuni** [comment]
@aweisberg : Can you please help get this reviewed?

**Comment by aweisberg** [comment]
Thanks for working on this. Is there some reasonable set of tests we could run on 3.x all the time? It would be good to have non-zero coverage without exploding the test run time.

It's better to not merge together all the commits because it makes it hard to logically review each change. It's possible that will be a blocker for the committer although this change set is small enough that I don't personally mind.

File: presto-hive-hadoop2/src/test/java/com/facebook/presto/hive/TestHiveClient.java
Comment (aweisberg) [inline presto-hive-hadoop2/src/test/java/com/facebook/presto/hive/TestHiveClient.java:102]: What is the nature of this failure that is making us exclude it?

File: presto-hive-hadoop2/bin/common.sh
Comment (aweisberg) [inline presto-hive-hadoop2/bin/common.sh:49]: Why was this necessary? I don't see it in Trino.

**Review by aweisberg — COMMENTED** [review]
Two questions

File: presto-hive-hadoop2/src/test/java/com/facebook/presto/hive/TestHiveClient.java
Comment (nmahadevuni) [reply presto-hive-hadoop2/src/test/java/com/facebook/presto/hive/TestHiveClient.java:102]: The varchar to integer coercion fails in Hive >=3.

File: presto-hive-hadoop2/bin/common.sh
Comment (nmahadevuni) [reply presto-hive-hadoop2/bin/common.sh:49]: These two services are started only for Hive < 3. For the new Hive 3.1.2 image, these services won't be started, so trying to stop them throws an error. My guess is Trino is not using Apache Hive 3.1.2 image for tests. Otherwise, they would have faced this problem too.

**Comment by nmahadevuni** [comment]
> Thanks for working on this. Is there some reasonable set of tests we could run on 3.x all the time? It would be good to have non-zero coverage without exploding the test run time.
> 
> It's better to not merge together all the commits because it makes it hard to logically review each change. It's possible that will be a blocker for the committer although this change set is small enough that I don't personally mind.

The number of tests are not too high. I see that with the current HDP2.6 Hive, the tests, mainly setup(DDL and data loading) take a very long time. The tests on Hive 3.1.2 version run extremely fast probably because the execution engine is changed to Tez. 

When a related PR is created that needs tests to be run on both versions, we can ask the developers to run on local with both versions. Can we add a new workflow where once the changes are approved, it will kick off tests on both versions before the PR is merged?

**Comment by nmahadevuni** [comment]
@aweisberg : Anymore comments on this? Can you please add another reviewer?

**Review by aweisberg — APPROVED** [review]

**Comment by aweisberg** [comment]
@nmahadevuni one bit of feedback. Please don't merge commits when cherry-picking. Smaller commits are much easier to review when they contain one logical related change along with the explanation for why that change was required.

The easier something is to review the more likely the review gets done and completed before the reviewer gets pulled off to do something else :-)

**Review by highker — APPROVED** [review]

**Comment by highker** [comment]
FB test not triggered. Validated locally.

**Pull request merged**
